framework/src/DriverFramework.cpp

```diff
@@ -629,21 +629,10 @@ void HRTWorkQueue::process(void)
 
 		// pthread_cond_timedwait uses absolute time
 		ts = offsetTimeToAbsoluteTime(now + next);
+		DF_LOG_DEBUG("HRTWorkQueue::process waiting for work (%" PRIi64 ")", diff);
+		// Wait until next expiry or until a new item is rescheduled
+		pthread_cond_timedwait(&g_reschedule_cond, &g_hrt_lock, &ts);
```

Comment:
i see the previous code wasn't, but would be good to check the return value from `pthread_cond_timedwait()` to verify that it's `0` or `ETIMEDOUT` as assumed here.